CoseHandler/CoseHandler.cs

```diff
@@ -591,27 +592,54 @@ public static class CoseHandler
         if (publicKey is null)
         {
             errorCodes.Add(ValidationFailureCode.NoPublicKey);
-            return new ValidationResult(false, errorCodes, certValidationResults, chain);
+            return new ValidationResult(false, errorCodes, certValidationResults, chain, validationType: ContentValidationType.ContentValidationNotPerformed);
         }
 
         // Validate that the COSE header is formatted correctly and that the payload and hash are consistent.
         bool messageVerified = false;
+
+        // Checks for an indirect signature, where the content header contains the hash of the payload, and the algorithm is stored in the message.
+        // If this is the case, we need to make sure the external payload hash matches the hash stored in the cose message content.
+        bool indirectSignature = msg.IsDetachedSignature();
+        ContentValidationType validationType = ContentValidationType.ContentValidationNotPerformed;
+
         try
         {
             if (!payloadBytes.IsNullOrEmpty())

```

Comment:
I think you can simplify here by setting the ContentValidationType first.
`bool hasBytes = !payloadBytes.IsNullOrEmpty();
bool hasStream = payloadStream is not null;
ContentValidationType cvt = 
    indirectSignature ? ContentValidationType.Indirect : 
    hasBytes || hasStream ? ContentValidationType.Detached :
    ContentValidationType.Embedded;
switch cvt
    ...`

Reply:
this is so much cleaner, thank you!